server/functions.lua

```diff
@@ -315,18 +313,19 @@ function QBCore.Functions.AddPermission(source, permission)
 end
 
 function QBCore.Functions.RemovePermission(source, permission)
-    if permission then
-        if IsPlayerAceAllowed(source, permission) then
-            ExecuteCommand(('remove_principal player.%s qbcore.%s'):format(source, permission))
-            QBCore.Commands.Refresh(source)
-        end
-    else
+    if not permission then
         for _, v in pairs(QBCore.Config.Server.Permissions) do
             if IsPlayerAceAllowed(source, v) then
                 ExecuteCommand(('remove_principal player.%s qbcore.%s'):format(source, v))
                 QBCore.Commands.Refresh(source)
             end
         end
+        return
+    end
+    
+    if IsPlayerAceAllowed(source, permission) then
+        ExecuteCommand(('remove_principal player.%s qbcore.%s'):format(source, permission))
+        QBCore.Commands.Refresh(source)
```

Comment:
I think this function is less readable now.

Reply:
I was unhappy with this one as well, I'll revert it back to the original until i can find another way to tidy it up.